super-linter/super-linter
# Fix GitHub Actions debug logging

# Proposed changes

Enable debug logging when one (or more) of the following conditions is
true:

- ACTIONS_RUNNER_DEBUG is set to true
- ACTIONS_STEPS_DEBUG is set to true
- RUNNER_DEBUG is set to 1

To test this change, I added:

- Three new make targets: one for each of the variables mentioned above.
- One new build step in the `CI` workflow to test the local action without enabling debug logging. By re-running the workflow with GitHub Actions debug mode enabled, it shows debug messages as expected. Example: https://github.com/super-linter/super-linter/actions/runs/7921109078/job/21627279501?pr=5288#step:10

Remove `ACTIONS_RUNNER_DEBUG` from the README so we point users to `LOG_LEVEL` as the preferred mechanism to configure the log level.

## Readiness checklist

In order to have this pull request merged, complete the following tasks.

### Pull request author tasks

- [x] I included all the needed documentation for this change.
- [x] I provided the necessary tests.
- [x] I squashed all the commits into a single commit.
- [x] I followed the [Conventional Commit v1.0.0 spec](https://www.conventionalcommits.org/en/v1.0.0/).
- [x] I wrote the necessary upgrade instructions in the [upgrade guide](../docs/upgrade-guide.md).
- [x] If this pull request is about and existing issue,
  I added the `Fix #ISSUE_NUMBER` label to the description of the pull request.

### Super-linter maintainer tasks

- [x] Label as `breaking` if this change breaks compatibility with the previous released version.
- [x] Label as either: `automation`, `bug`, `documentation`, `enhancement`, `infrastructure`.
